pkgs/misc/cups/drivers/kyodialog/default.nix

```
@@ -0,0 +1,109 @@
+{ lib
+, stdenv
+, fetchzip
+, cups
+, autoPatchelfHook
+, python3Packages
+
+# Sets the default paper format: use "EU" for A4, or "Global" for Letter
+, region ? "Global"
+# optional GUI, quite redundant to CUPS admin web GUI
+, withQtGui ? false
+, qt5
+}:
+
+# Open issues:
+#
+# Printing in grayscale mode (e.g. the test page) generates a warning
+# > [Job 59] Grayscale/monochrome printing requested for this job but Poppler is not able to convert to grayscale/monochrome PostScript.
+# > [Job 59] Use \"pdftops-renderer\" option (see cups-filters README file) to use Ghostscript or MuPDF for the PDF -> PostScript conversion.
+#
+# This is related to https://github.com/OpenPrinting/cups-filters/issues/169.
+
+let
+  platform =
+    if stdenv.hostPlatform.system == "x86_64-linux" then "amd64"
+    else if stdenv.hostPlatform.system == "i686-linux" then "i386"
+         else throw "Unsupported system: ${stdenv.hostPlatform.system}";
+  kyodialog_version = "9.2-0";
+in
+stdenv.mkDerivation rec {
+  pname = "cups-kyodialog";
+  version = "${builtins.replaceStrings ["-"] ["."] kyodialog_version}-20220928";
+
+  dontStrip = true;
+
+  src = let
+    date = lib.lists.last (builtins.splitVersion version);
+  in fetchzip {
+    url = "https://www.kyoceradocumentsolutions.us/content/download-center-americas/us/drivers/drivers/KyoceraLinuxPackages_${date}_tar_gz.download.gz";
```

```suggestion
  src = fetchzip {
    url = "https://www.kyoceradocumentsolutions.us/content/download-center-americas/us/drivers/drivers/KyoceraLinuxPackages_${date}_tar_gz.download.gz";
```

I don't trust them to keep this download. Please upload it to web.archive.org.